src/dsp_tools/commands/excel2json/lists.py

```diff
@@ -41,37 +37,42 @@ def expand_lists_from_excel(
     Returns:
         the same "lists" section, but without references to Excel files
     """
-    new_lists = []
+    new_lists: list[dict[str, Any]] = []
     for _list in lists_section:
         # case 1: this list is a JSON list: return it as it is
         if "folder" not in _list["nodes"]:
             new_lists.append(_list)
             continue
 
         # case 2: this is a reference to a folder with Excel files
-        foldername = _list["nodes"]["folder"]  # type: ignore[index]  # types are too complex to annotate them correctly
```

Comment:
This extensive refactoring was necessary because sourcery rated the code quality as too low.